### PR TITLE
e2e: switch default test distro to fedora/40-cloud-base.

### PR DESCRIPTION
**Pull request by klihub**
Fedora 37 is EOL. Switch default test distro to fedora 40 and use cloud base image.

### CODE REVIEW

**Comment by klihub** [comment]
@askervin @fmuyassarov @kad @pfl PTAL.

**Review by pfl — APPROVED** [review]

**Review by kad — APPROVED** [review]
LGTM

File: test/e2e/playbook/provision.yaml [hunk 1]
```diff
@@ -139,13 +139,23 @@
...
+            pkg:
+              - iproute-tc
+              - grubby
+            state: present
+            update_cache: true
+        - name: Extra python3 rpm packages for ansible
+          ansible.builtin.dnf:
+            pkg:
+              - python3-libselinux
+              - python3-packaging
```
Comment (fmuyassarov) [inline test/e2e/playbook/provision.yaml:155]: I would merge this two tasks into one and create a single list of requires packages.

**Review by fmuyassarov — APPROVED** [review]

File: test/e2e/playbook/provision.yaml (hunk 1 above)
Comment (klihub) [reply test/e2e/playbook/provision.yaml:155]: The 2nd one is conditional not necessary (and neither possible) on our older Fedora distros.

**Pull request merged**